cxx-sensors/src/main/java/org/sonar/cxx/sensors/coverage/TestwellCtcParser.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Sonar C++ Plugin (Community)
+ * Copyright (C) 2010-2018 SonarOpenCommunity
+ * http://github.com/SonarOpenCommunity/sonar-cxx
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.cxx.sensors.coverage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+import java.util.NoSuchElementException;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+
+import java.math.BigDecimal;
+
+import org.apache.commons.io.FilenameUtils;
+
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.FILE_HEADER;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.FILE_RESULT;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.LINE_RESULT;
+import static org.sonar.cxx.sensors.coverage.TestwellCtcResult.SECTION_SEP;
+
+
+
+/**
+ * {@inheritDoc}
+ */
+public class TestwellCtcParser extends CxxCoverageParser {
+
+  private static final Logger LOG = Loggers.get(TestwellCtcParser.class);
+  
+  private Scanner scanner;
+  private Matcher matcher;
+  private State state;
+  
+  
+  private static final int FROM_START = 0;
+  private static final int CONDS_FALSE = 1;
+  private static final int CONDS_TRUE = 2;
+  private static final int LINE_NR_GROUP = 3;
+  
+  private enum State {
+    BEGIN,
+    PARSING,
+    END;
+  }
+  
+
+  public TestwellCtcParser() {
+    // no operation but necessary for list of coverage parsers 
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void processReport(final SensorContext context, File report, final Map<String, CoverageMeasures> coverageData) {
+    LOG.debug("Parsing 'Testwell CTC++' format");
```

Comment:
To make clear that ‚only‘ textual format is supported:
Parsing 'Testwell CTC++' **textual** format